git-server/src/lib.rs

```diff
@@ -305,54 +318,68 @@ pub async fn run(options: Options) -> anyhow::Result<()> {
         bail!("Failed to disable gc: {:?}", e);
     }
 
-    let peer_id = path::param::<PeerId>()
-        .map(Some)
-        .or_else(|_| async { Ok::<(Option<PeerId>,), Infallible>((None,)) });
-
-    let server = warp::filters::any::any()
-        .map(move || ctx.clone())
-        .and(warp::method())
-        .and(warp::filters::header::headers_cloned())
-        .and(warp::filters::body::aggregate())
-        .and(warp::filters::addr::remote())
-        .and(path::param())
-        .and(peer_id)
-        .and(path::peek())
-        .and(
-            warp::filters::query::raw()
-                .or(warp::any().map(String::default))
-                .unify(),
+    let app = git_handler
+        .layer(Extension(ctx.clone()))
+        .layer(
+            TraceLayer::new_for_http()
+                .make_span_with(DefaultMakeSpan::new().level(Level::INFO))
+                .on_response(
+                    |response: &Response<BoxBody>, latency: Duration, _span: &Span| {
+                        tracing::info!("STATUS={:?} LATENCY={:?}", response.status(), latency)
+                    },
+                ),
         )
-        .and_then(git_handler)
-        .recover(recover)
-        .with(warp::log("radicle_git_server"));
-    let server = warp::serve(server);
+        .into_make_service_with_connect_info::<SocketAddr>();
 
     if let (Some(cert), Some(key)) = (options.tls_cert, options.tls_key) {
-        server
-            .tls()
-            .cert_path(cert)
-            .key_path(key)
-            .run(options.listen)
-            .await;
+        let config = RustlsConfig::from_pem_file(cert, key).await.unwrap();
+
+        tracing::info!("listening on https://{}", options.listen);
+        axum_server::bind_rustls(options.listen, config)
+            .serve(app)
+            .await
+            .unwrap();
     } else {
-        server.run(options.listen).await;
+        tracing::info!("listening on http://{}", options.listen);
+        axum::Server::bind(&options.listen)
+            .serve(app)
+            .await
+            .unwrap();
```

Comment:
Can we return errors instead of unwrapping in this func?

Reply:
https://github.com/radicle-dev/radicle-client-services/compare/221ab2a75a7b7774964a64eded3f510cca480713..db63e7a4f28c0d0e7beb5b1a1cf2339d6129722d#diff-b672a15ad14f54528cebd77aee76d3007962d44bda987b3e0a34597ec80fd6b1R352-R355